plugins/CoreHome/javascripts/dataTable.js

```diff
@@ -1377,7 +1377,12 @@ $.extend(DataTable.prototype, UIControl.prototype, {
 
     getFilterParams: function (params) {
         return Object.keys(params)
-           .filter(key => key.startsWith('filter_column') || key.startsWith('filter_pattern'))
+           .filter(key => {
+               return key === 'filter_trigger_id'
+                   || key.startsWith('filter_column')
+                   || key.startsWith('filter_pattern')
+                   || key.startsWith('filter_sort')
```

Comment:
This should not be needed for table reports, as the sorting is already stored using `saveViewDataTableParameters`.
However this is not done for dashboard widgets, they are currently not remembering that information on purpose. If storing the sorting for dashboard widgets is a general desired behavior this maybe should be stored in the dashboard layout instead to ensure it persists on reload.

Reply:
I agree that it's strange that reports behave differently between normal report pages and the dashboard.
But if we want to change the behavior for dashboard widgets I would suggest to do that for all cases and not only if you search. So be better to handle that in a different issue / PR.